### PR TITLE
Mariuszhussariush/ch7763/create embeddable additional cost calculator

### PR DESCRIPTION
**Pull request by mariusz-hussariush**
initial commit, state handling from form, markup

### CODE REVIEW

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
+import { Component, h, State } from '@stencil/core';
+// import { format } from '../../utils/utils';
```
Comment (hariscustovic) [inline additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx]: can we remove the comment?

File: additionalcostcalculator/src/index.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0">
+  <title>Stencil Component Starter</title>
+
+  <script type="module" src="/build/additionalcostcalculator.esm.js"></script>
+  <script nomodule src="/build/additionalcostcalculator.js"></script>
+
+</head>
+<body>
+
+  <additional-cost-calculator first="Stencil" last="'Don't call me a framework' JS"></additional-cost-calculator>
```
Comment (hariscustovic) [inline additionalcostcalculator/src/index.html]: we can remove the attributes?

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.e2e.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  it('renders changes to the name data', async () => {
+    const page = await newE2EPage();
+
+    await page.setContent('<additional-cost-calculator></additional-cost-calculator>');
+    const component = await page.find('additional-cost-calculator');
+    const element = await page.find('additional-cost-calculator >>> div');
+    expect(element.textContent).toEqual(`Hello, World! I'm `);
+
+    component.setProperty('first', 'James');
```
Comment (hariscustovic) [inline additionalcostcalculator/src/components/my-component/additional-cost-calculator.e2e.ts]: maybe we should comment out the fields we don't really use?

File: additionalcostcalculator/src/utils/utils.spec.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { format } from './utils';
+
+describe('format', () => {
+  it('returns empty string for no names defined', () => {
+    expect(format(undefined, undefined, undefined)).toEqual('');
+  });
+
+  it('formats just first names', () => {
```
Comment (hariscustovic) [inline additionalcostcalculator/src/utils/utils.spec.ts]: I would maybe comment this one as well.

Comment (hariscustovic) [reply additionalcostcalculator/src/utils/utils.spec.ts]: If it's not used, maybe we can even ditch it?

File: additionalcostcalculator/src/utils/utils.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (hariscustovic) [inline additionalcostcalculator/src/utils/utils.ts]: maybe we don't need this method at all?

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx (hunk 1 above)
Comment (mariusz-hussariush) [reply additionalcostcalculator/src/components/my-component/additional-cost-calculator.tsx]: done

File: additionalcostcalculator/src/index.html (hunk 2 above)
Comment (mariusz-hussariush) [reply additionalcostcalculator/src/index.html]: done

File: additionalcostcalculator/src/components/my-component/additional-cost-calculator.e2e.ts (hunk 3 above)
Comment (mariusz-hussariush) [reply additionalcostcalculator/src/components/my-component/additional-cost-calculator.e2e.ts]: done

File: additionalcostcalculator/src/utils/utils.spec.ts (hunk 4 above)
Comment (mariusz-hussariush) [reply additionalcostcalculator/src/utils/utils.spec.ts]: i just left  the rendered test

File: additionalcostcalculator/src/utils/utils.ts (hunk 5 above)
Comment (mariusz-hussariush) [reply additionalcostcalculator/src/utils/utils.ts]: true

**Pull request merged**